tests/integration/records/unload-test.js

```diff
@@ -419,3 +419,48 @@ test('unloading a disconnected subgraph clears the relevant internal models', fu
 
   assert.equal(relPayloads.get('person', 1, 'cars'), null, 'person - cars relationship payload unloaded');
 });
+
+
+test("after unloading a record, the record can be fetched again", function(assert) {
+  const store = env.store;
+  let record;
+
+  // stub findRecord
+  env.adapter.findRecord = () => {
+    return Ember.RSVP.Promise.resolve({
+      data: {
+        type: 'person',
+        id: '1',
+        attributes: {
+          name: 'Adam Sunderland'
+        }
+      }
+    });
+  };
+
+  // populate initial record
+  run(function() {
+    record = store.push({
+      data: {
+        type: 'person',
+        id: '1',
+        attributes: {
+          name: 'Adam Sunderland'
+        }
+      }
+    });
+  });
+
+  const internalModel = record._internalModel;
+  assert.equal(internalModel.currentState.stateName, 'root.loaded.saved', 'We are loaded initially');
+
+  // we test that we can sync call unloadRecord followed by findRecord
+  run(function() {
+    store.unloadRecord(record);
+    assert.equal(record.isDestroying, true, 'the record is destroying');
+    assert.equal(internalModel.currentState.stateName, 'root.empty', 'We are unloaded after unloadRecord');
+    store.findRecord('person', '1');
```

Comment:
this should likely cancel the above `unloadRecords`'s `_checkForOrphanedInternalModels` and unset `_isDematerializing`, so it can be back in a pristine place, ready for a findRecord